### PR TITLE
Solutions part 3

### PR DESCRIPTION
**Pull request by nataliadombrowska**
Run Length Encoding 
Given a string of characters return the run length encoded version. E.g. given the string “aabbbccddddaaa”, the return value would be “a2b3c2d4a3”.

Longest Substring
Write a function that takes a string and returns the longest substring in that string made up of the same character and the index the substring starts on, e.g. given the string “aabbbccddddaaa” the substring is “dddd” starting at index 7.

Second Lowest Number
Write a function that takes an array of numbers and returns the second lowest number in the array, or zero if the length of the array is less than two.

### CODE REVIEW

File: src/PairSumming.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Test
+    public void test3() {
+        Integer[] arr = {-1,2,1};
+        int actual = pairSumming(arr, 1);
+        int expected = 1;
+        assertEquals(expected, actual);
+    }
+
```
Comment (pling-scottlogic) [inline src/PairSumming.java:52]: Try this test case: `{1, 3, 2, 2, 3, 2}`

File: src/RunLengthEncoding.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void test4() {
+        String actual = runLengthEncoding("aabbbc");
+        String expected = "a2b3c1";
+        assertEquals(expected, actual);
+    }
+}
```
Comment (pling-scottlogic) [inline src/RunLengthEncoding.java:64]: Try a test case with a longer run at the end, e.g. `"aabbcc"`

File: src/SingleCharacterSearch.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                char currentLetter = word.charAt(i);
+                if(characterFrequency.containsKey(currentLetter)){
+                    characterFrequency.put(currentLetter, characterFrequency.get(word.charAt(i)) + 1);
+                } else {
+                    characterFrequency.put(currentLetter, 1);
+                }
+            }
+
+            //loop through hashmap and return the first character in the map that has value of 1
+            for(Character letter: characterFrequency.keySet()){
```
Comment (pling-scottlogic) [inline src/SingleCharacterSearch.java:31]: I don't think hash map key sets are ordered. You're probably better off looping through the input word again to make sure you get the first letter that only appears once. It's not easy to write a test case for this though.

**Review by pling-scottlogic — COMMENTED** [review]
Good answers, and well presented. A few edge cases to consider though.
